README.md

```diff
@@ -257,12 +257,28 @@ A value that implements the Traversable specification must also
 implement the Functor specification.
 
 1. `t(u.sequence(f.of))` is equivalent to `u.map(t).sequence(g.of)`
-where `t` is a natural transformation from `f` to `g` (naturality)
-
-2. `u.map(x => Id(x)).sequence(Id.of)` is equivalent to `Id.of(u)` (identity)
-
-3. `u.map(Compose).sequence(Compose.of)` is equivalent to
-   `Compose(u.sequence(f.of).map(x => x.sequence(g.of)))` (composition)
+for any `t` such that `t(x).map(a)` is equivalent to `t(x.map(a))` (naturality)
+
+2. `u.map(F.of).sequence(F.of)` is equivalent to `F.of(u)` for any Applicative `F` (identity)
+
+3. `u.map(Compose.of).sequence(Compose.of)` is equivalent to
+   `Compose.of(u.sequence(f.of).map(x => x.sequence(g.of)))` for Compose type defined below (composition)
+
+```js
+Compose = {
+  of(c) {
+    return {
+      c,
+      map(f) {
+        return Compose.of(c.map(y => y.map(f)))
+      },
+      ap(x) {
+        return Compose.of(c.map(u => y => u.ap(y)).ap(x.c))
+      }
+    }
+  }
+}
```

Comment:
Prototype, as that's how `Id` is implemented, we should at least be
consistent.

On Sun, 27 Mar 2016, 17:36 Roman Pominov, <notifications@github.com> wrote:

> In README.md
> <https://github.com/fantasyland/fantasy-land/pull/131#discussion_r57531273>
> :
>
> > +   `Compose.of(u.sequence(f.of).map(x => x.sequence(g.of)))` for Compose type defined below (composition)
> > +
> > +```js
> > +Compose = {
> > +  of(c) {
> > +    return {
> > +      c,
> > +      map(f) {
> > +        return Compose.of(c.map(y => y.map(f)))
> > +      },
> > +      ap(x) {
> > +        return Compose.of(c.map(u => y => u.ap(y)).ap(x.c))
> > +      }
> > +    }
> > +  }
> > +}
>
> Is this code style ok for this code? Do we want to use ES6? Should we
> rewrite it with prototype or ES6 class maybe?
>
> I also not 100% sure this implementation is correct. Will double check
> later.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/fantasyland/fantasy-land/pull/131/files/3145795b424e382fea7dcb0117e75e9b8e4e9df8#r57531273>
>